Sources/MacroToolkit/Type.swift

```diff
@@ -174,6 +169,199 @@ public enum `Type`: TypeProtocol, SyntaxExpressibleByStringInterpolation {
     }
 
     // TODO: Implement rest of conversions
+    
+    public func normalized() -> NormalizedType {
+        switch self {
+        case .array(let type):
+            var arrayTypeSyntax: ArrayTypeSyntax = type._baseSyntax
+            var attributedTypeSyntax: AttributedTypeSyntax? = type._attributedSyntax
+            let normalizedElement = Type(arrayTypeSyntax.element).normalized()
+            arrayTypeSyntax.element = TypeSyntax(normalizedElement._syntax)
+            attributedTypeSyntax?.baseType = TypeSyntax(arrayTypeSyntax)
+            
+            var base = "Array<\(arrayTypeSyntax.element)>"
+            if let attributedTypeSyntax {
+                base = base.addingAttributes(from: attributedTypeSyntax)
+            }
+            return NormalizedType(stringLiteral: base)
+        case .classRestriction(let type):
+            // Not handling `_attributedSyntax` because `classRestriction` cannot have any attribute
+            
+            // let normalizedType: NormalizedType = "AnyObject"
+            let normalizedType: NormalizedType = .simple(.init(.init(
+                leadingTrivia: type._baseSyntax.leadingTrivia,
+                name: .identifier("AnyObject"),
+                trailingTrivia: type._baseSyntax.trailingTrivia
+            )))
+            return normalizedType
+            
+        case .composition(let type):
+            // Looks like there can only be simple types in composition, with no generics, and therefore we
```

Comment:
How is syntax such as `(Encodable & Sequence<Int?>) & Equatable` parsed? I assume it gets interpreted as something like `(Encodable & Sequence<Int?>) & Equatable` in which case you still need to continue recursively normalizing to reach `Int?`.